### ISSUE TITLE
Incompatible with V8 4.1.x (io.js)

### ISSUE THREAD

**Issue by kbanman** [brianc/node-postgres#763]
When trying to install using io.js 1.6.4, compilation fails:
```
> pg@3.4.5 install node_modules/pg
> node-gyp rebuild || (exit 0)

make: Entering directory 'node_modules/pg/build'
  CXX(target) Release/obj.target/binding/src/binding.o
In file included from ../src/binding.cc:3:0:
../node_modules/nan/nan.h:481:19: error: ‘NanNew’ declared as an ‘inline’ variable
       v8::String::ExternalAsciiStringResource *resource) {
                   ^
../node_modules/nan/nan.h:481:19: warning: ‘always_inline’ attribute ignored [-Wattributes]
../node_modules/nan/nan.h:481:19: error: ‘v8::Local<v8::String> NanNew’ redeclared as different kind of symbol
../node_modules/nan/nan.h:475:36: note: previous declaration ‘v8::Local<v8::String> NanNew(v8::String::ExternalStringResource*)’
   NAN_INLINE v8::Local<v8::String> NanNew(
                                    ^
../node_modules/nan/nan.h:481:7: error: ‘ExternalAsciiStringResource’ is not a member of ‘v8::String’
       v8::String::ExternalAsciiStringResource *resource) {
       ^
../node_modules/nan/nan.h:481:48: error: ‘resource’ was not declared in this scope
       v8::String::ExternalAsciiStringResource *resource) {
                                                ^
../node_modules/nan/nan.h: In function ‘bool _NanGetExternalParts(v8::Handle<v8::Value>, const char**, size_t*)’:
../node_modules/nan/nan.h:2103:12: error: ‘class v8::String’ has no member named ‘IsExternalAscii’
   if (str->IsExternalAscii()) {
            ^
../node_modules/nan/nan.h:2104:23: error: ‘ExternalAsciiStringResource’ in ‘class v8::String’ does not name a type
     const v8::String::ExternalAsciiStringResource* ext;
                       ^
../node_modules/nan/nan.h:2105:5: error: ‘ext’ was not declared in this scope
     ext = str->GetExternalAsciiStringResource();
     ^
../node_modules/nan/nan.h:2105:16: error: ‘class v8::String’ has no member named ‘GetExternalAsciiStringResource’
     ext = str->GetExternalAsciiStringResource();
                ^
binding.target.mk:88: recipe for target 'Release/obj.target/binding/src/binding.o' failed
make: *** [Release/obj.target/binding/src/binding.o] Error 1
make: Leaving directory 'node_modules/pg/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1044:12)
gyp ERR! System Linux 3.16.0-33-generic
gyp ERR! command "/usr/bin/iojs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd node_modules/pg
gyp ERR! node -v v1.6.4
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok 
pg@3.4.5 node_modules/pg
├── bindings@1.2.1
├── packet-reader@0.2.0
├── pg-connection-string@0.1.3
├── buffer-writer@1.0.0
├── generic-pool@2.1.1
├── pg-types@1.4.0
├── nan@1.3.0
└── pgpass@0.0.3 (split@0.3.3)
```
I realize io.js is not listed as a supported platform, but I expect node.js will also be moving to newer versions of V8, so it would be helpful to fix.

**Comment by vitaly-t** [comment]
Seems like a very generic issue with IO.js. If you google for the errors, they are reported everywhere.

In the meantime, PG seems to install and work just fine under the latest NodeJS 0.12.2

**Comment by vitaly-t** [comment]
@kbanman , are you still getting the same error with IO.js 1.7.1?

Is this a 64-bit IO.js under being run under Linux?

**Comment by brianc** [comment]
If someone with familiarity with node-nan would like to take the time to upgrade node-pg-native to support IO.js it should be pretty straight-forward.  I don't use IO.js nor do I have it set up on my system right now, so it will take me a while to get to it. https://github.com/brianc/node-pg-native

**Comment by litek** [comment]
Are you installing pg 3.4.5? Latest version seems to be working fine with iojs.

**Comment by vitaly-t** [comment]
@litek 
>Are you installing pg 3.4.5? Latest version seems to be working fine with iojs.

The latest version is 4.3, so what are you talking about? :)

**Comment by litek** [comment]
The log mentions pg@3.4.5, and 4.3.0 works fine here with the latest iojs (Linux/OS X x64).

```
> pg@3.4.5 install node_modules/pg
pg@3.4.5 node_modules/pg
```

**Comment by vitaly-t** [comment]
@litek , wow, well spotted, I didn't even notice that! :+1: So, we just might be in the clear then :) Let's wait for @kbanman to confirm that 4.3 works ;)

**Comment by brianc** [comment]
Does installing the native bindings also work? `npm i pg-native` ? The pure
javascript bindings should work with no problems for sure...there is some
compile time dependency stuff w/ IO.js I'm not sure we've worked out yet.

**Comment by litek** [comment]
Yeah, latest pg-native work as well.

**Comment by vitaly-t** [comment]
Excellent. I'm gonna add IO.js testing profile for Travis CI to pg-promise, and it will be great if Brian does the same for PG ;)

**Comment by vitaly-t** [comment]
@brianc , I've just created a pull request for PG to include IO.js into the test.

Just so, prior to this I did the same for pg-promise, and all tests passed.

**Comment by vitaly-t** [comment]
P.P.S. And you can throw in:

![io.js supported](https://img.shields.io/badge/io.js-supported-green.svg)

**Comment by vitaly-t** [comment]
@brianc, takes only 1 min to push those changes ;)

**Comment by jonathanong** [comment]
for sure, you need to update nan's libpq. 1.5.0 is super broken and you've pinned it so we can't not use a broken version

**Comment by brianc** [comment]
sorry for taking a while to get to this. Merged the badge & also @jonathanong I've unpinned pg-native & libpq from nan - it now installs the most recent 1.x version of node-nan so hopefully this fixes your issue?

**Comment by brianc** [comment]
available in libpq@1.6.0 which installs by default w/ a new install of pg-native

Result of `npm ls` after doing `npm i pg-native` locally:
```
└─┬ pg-native@1.8.0
  ├─┬ libpq@1.6.0
  │ ├── bindings@1.2.1
  │ └── nan@1.8.4
  ├── pg-types@1.6.0
  └─┬ readable-stream@1.0.31
    ├── core-util-is@1.0.1
    ├── inherits@2.0.1
    ├── isarray@0.0.1
    └── string_decoder@0.10.31
```
